larray/core/array.py

```diff
@@ -10388,23 +6734,16 @@ def read_excel(filepath, sheetname=0, nb_index=None, index_col=None, na=np.nan,
     if engine is None:
         engine = 'xlwings' if xw is not None else None
 
-    if nb_index is not None and index_col is not None:
-        raise ValueError("cannot specify both nb_index and index_col")
-    elif nb_index is not None:
-        index_col = list(range(nb_index))
-    elif isinstance(index_col, int):
-        index_col = [index_col]
-
     if engine == 'xlwings':
         if kwargs:
             raise TypeError("'{}' is an invalid keyword argument for this function when using the xlwings backend"
                             .format(list(kwargs.keys())[0]))
-        from .excel import open_excel
+        from larray.io.excel import open_excel
         with open_excel(filepath) as wb:
-            return wb[sheetname].load(index_col=index_col)
+            return wb[sheetname].load(index_col=_get_index_col(nb_index, index_col))
     else:
-        df = pd.read_excel(filepath, sheetname, index_col=index_col, engine=engine, **kwargs)
-        return df_aslarray(df, sort_rows=sort_rows, sort_columns=sort_columns, raw=index_col is None, fill_value=na)
+        df, raw = _read_excel(filepath, sheetname, nb_index, index_col, engine, **kwargs)
```

Comment:
this PR is already hard enough to review as it is, could you move this change to a separate PR (possibly depending on this one)